Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DriverOnboardingV2.hs

```diff
@@ -1078,35 +1123,320 @@ postDriverRegisterCommonDocument (mbDriverId, merchantId, merchantOperatingCityI
-      then do
-        mbFleetDriverAssociation <- FDA.findByDriverId driverId True
-        case mbFleetDriverAssociation of
-          Nothing -> pure []
-          Just fleetDriverAssociation -> do
-            let fleetOwnerId = fleetDriverAssociation.fleetOwnerId
-            RCQuery.findAllByFleetOwnerId effectiveLimit offset (Just fleetOwnerId)
-      else pure []
-  let allRcs = DL.nubBy (\a b -> a.id == b.id) (driverRcs <> fleetRcs)
-      activeRcId = fmap (.rcId) . DL.find (.isRcActive) $ driverLinkedRcs
-  rcs <- mapM (getCombinedRcData activeRcId) allRcs
-  return $ APITypes.FleetRCListRes rcs
-  where
-    getCombinedRcData activeRcId rc = do
-      rcNo <- decrypt rc.certificateNumber
-      let isActive = Just rc.id == activeRcId
-      return $
-        VRC.LinkedRC
-          { rcActive = isActive,
-            rcDetails = SDO.makeRCAPIEntity rc rcNo,
-            isFleetRC = isJust rc.fleetOwnerId
+  logInfo $ "DigiLocker initiate - Starting authorization flow for DriverId: " <> driverId.getId <> ", VehicleCategory: " <> show req.vehicleCategory
+
+  -- Validate vehicle category - only auto, car, and bike are allowed
+  let allowedVehicleCategories = [DVC.AUTO_CATEGORY, DVC.CAR, DVC.MOTORCYCLE]
+  unless (req.vehicleCategory `elem` allowedVehicleCategories) $
+    throwError $ InvalidRequest $ "Vehicle category must be one of: auto, car, or bike. Received: " <> show req.vehicleCategory
+
+  -- Step 1: Verify DigiLocker is enabled for this merchant+city
+  DigilockerLockerShared.verifyDigiLockerEnabled merchantOpCityId
+
+  -- Step 2: Check for existing active session
+  latestSession <- QDV.findLatestByDriverId (Just 1) (Just 0) driverId
+
+  case latestSession of
+    [] -> do
+      -- No existing session - create new one
+      logInfo $ "DigiLocker initiate - No existing session found for DriverId: " <> driverId.getId
+      createNewDigiLockerSession driverId merchantId merchantOpCityId req.vehicleCategory
+    (session : _) -> do
+      -- Existing session found - validate and decide action
+      logInfo $ "DigiLocker initiate - Found existing session for DriverId: " <> driverId.getId <> ", SessionStatus: " <> show session.sessionStatus
+      handleExistingSession driverId merchantId merchantOpCityId session req.vehicleCategory
+
+----------- Helper Functions for DigiLocker Initiate -----------
+
+-- Handle existing session logic
+handleExistingSession ::
+  Id Domain.Types.Person.Person ->
+  Id Domain.Types.Merchant.Merchant ->
+  Id Domain.Types.MerchantOperatingCity.MerchantOperatingCity ->
+  DDV.DigilockerVerification ->
+  DVC.VehicleCategory -> -- Vehicle category for new session if needed
+  Flow APITypes.DigiLockerInitiateResp
+handleExistingSession driverId merchantId merchantOpCityId session vehicleCategory = do
+  now <- getCurrentTime
+
+  case session.sessionStatus of
+    DDV.PENDING -> handlePendingSession driverId merchantId merchantOpCityId session now
```

Comment:
what if user goes back from the digilocker screen after initiate

Reply:
@rohan-goel please send webpage URL in status API response. Create a function in sharedLogic to construct URL and call it both from statusHandler and Initiate API. You can use the logic of constructing URL from Initiate API